Packs/EclecticIQ/Integrations/EclecticIQV2/EclecticIQV2.py

```diff
@@ -0,0 +1,656 @@
+        )
+
+    def get_user_granted_permissions(self) -> Any:
+        """Get user granted permissions.
+        :param: self
+        :type: str
+        :return: user granted permissions
+        :rtype: Any
+        """
+        response = self._http_request(
+            method='GET',
+            url_suffix='users/self',
+            params={}
+        )
+        data = response.get("data")
+        if data:
+            return data.get("permissions")
+        return {}
+
+    def get_platform_permissions(self) -> Any:
+        """Get platform permissions for user.
+        :param: self
+        :type: str
+        :return: permissions data
+        :rtype: Any
+        """
+        response = self._http_request(
+            method='GET',
+            url_suffix='permissions',
+            params={}
+        )
+        data = response.get("data", {})
+        return data or {}
+
+
+def get_platform_permission_ids(permissions_data: Any) -> List[Any]:
+    """Get permission ids required for user to authenticate.
+    :param feeds: permissions_data
+    :type response: list
+        [{"id": 1, "name": "read history-events"},{"id": 2,"name": "read discovery-rules"}...]
+    :return: List of permission ids
+        [33, 59, 66,78]
+    :rtype: list
+    """
+    wanted_permissions = [
+        "read entitites",
+        "modify entities",
+        "read extracts",
+        "read outgoing-feeds",
+    ]
+    ids_required_for_user = []
+    for value in permissions_data:
+        if value.get("name") in wanted_permissions:
+            ids_required_for_user.append(value.get("id"))
+
+    return ids_required_for_user
+
+
+def authenticate_user(ids_of_user: list, ids_required_for_user: list) -> Tuple[bool, List[int]]:
+    """Get user authentication and missing permission ids .
+    :param ids_of_user: permission ids user have
+    :type ids_of_user: list
+    :param ids_required_for_user: permission ids required for user to authenticate
+    :type ids_required_for_user: list
+    :return: is user authenticated , missing permissions ids
+    :rtype: boolean,list
+    """
+    user_authenticated = False
+    value = list(set(ids_required_for_user).difference(ids_of_user))
+
+    if not value:
+        user_authenticated = True
+    return user_authenticated, value
+
+
+def get_permission_name_from_id(permission_data: Dict, permission_ids: list) -> Any:
+    """Get permission name from permission ids.
+    :param permission_data: permission data
+    :type permission_data: Dict
+    :param permission_ids: permission id for authenticate
+    :type permission_ids: list
+    :return: permissions name
+    :rtype: Any
+    """
+    permissions_name = []
+    for data in permission_data:
+        for permission_id in permission_ids:
+            if data.get("id") == permission_id:
+                permissions_name.append(data.get("name"))
+    return permissions_name
+
+
+def data_ingestion(client: Client) -> Any:
```

Comment:
It is better practice to call this function test_module but that is up to you.

Reply:
Unit testcases are giving error while using this function as "test_module", that's why we changed this function name.

Reply:
That sound a bit weird, I would love to see the error.
In any case it is not preventing us from moving forward.